### PR TITLE
never restrict options requests

### PR DESCRIPTION
**Pull request by kwypchlo**
preflight requests should be able to respond without requiring authentication

### CODE REVIEW

File: docker/nginx/libs/skynet/account.lua [hunk 1]
```diff
@@ -82,16 +82,26 @@ function _M.is_subscription_required()
...
+function is_access_always_allowed()
+    -- options requests do not attach cookies - should always be available
+    -- requests should not be limited based on accounts if accounts are not enabled
+    return ngx.req.get_method() == "OPTIONS" or not _M.accounts_enabled()
+end
+
+-- check whether access is restricted if portal requires authorization
 function _M.is_access_unauthorized()
-    return _M.accounts_enabled() and _M.is_auth_required() and not _M.is_authenticated()
+    if is_access_always_allowed() then return true end
```
Comment (ChrisSchinnerl) [inline docker/nginx/libs/skynet/account.lua]: `false`?

File: docker/nginx/libs/skynet/account.lua [hunk 2]
```diff
@@ -82,16 +82,26 @@ function _M.is_subscription_required()
...
+    -- check if authentication is required and request is not authenticated
+    return _M.is_auth_required() and not _M.is_authenticated()
 end
 
--- check whether access to portal should be restricted to users with active subscription
--- based on the configurable environment variable
+-- check whether user is authenticated but does not have access to given resources
 function _M.is_access_forbidden()
-    return _M.accounts_enabled() and _M.is_subscription_required() and not _M.is_subscription_account()
+    if is_access_always_allowed() then return true end
```
Comment (ChrisSchinnerl) [inline docker/nginx/libs/skynet/account.lua]: `false` ?

**Review by ChrisSchinnerl — CHANGES_REQUESTED** [review]

File: docker/nginx/libs/skynet/account.lua (hunk 1 above)
Comment (kwypchlo) [reply docker/nginx/libs/skynet/account.lua]: this makes me sad

File: docker/nginx/libs/skynet/account.lua (hunk 2 above)
Comment (kwypchlo) [reply docker/nginx/libs/skynet/account.lua]: this too

**Review by ro-tex — APPROVED** [review]

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**